subgraphs/rocket-pool/src/entityUpdates/financialMetrics.ts

```diff
@@ -0,0 +1,440 @@
+  stakingRewards: BigDecimal,
+  totalShares: BigInt // of rETH
+): void {
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  const protocol = getOrCreateProtocol();
+  const financialMetrics = getOrCreateFinancialDailyMetrics(block);
+  const poolMetricsDailySnapshot = getOrCreatePoolsDailySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+  const poolMetricsHourlySnapshot = getOrCreatePoolsHourlySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+
+  // Staking Rewards
+
+  // Pool
+  pool.cumulativeTotalRevenueUSD = stakingRewards;
+  pool.outputTokenSupply = totalShares;
+  pool.outputTokenPriceUSD = getOrCreateToken(
+    Address.fromString(PROTOCOL_ID),
+    block.number
+  ).lastPriceUSD;
+  pool.save();
+
+  // Pool Daily
+  poolMetricsDailySnapshot.cumulativeTotalRevenueUSD =
+    pool.cumulativeTotalRevenueUSD;
+  poolMetricsDailySnapshot.dailyTotalRevenueUSD = stakingRewards;
+  poolMetricsDailySnapshot.outputTokenSupply = pool.outputTokenSupply;
+  poolMetricsDailySnapshot.outputTokenPriceUSD = pool.outputTokenPriceUSD;
+  poolMetricsDailySnapshot.save();
+
+  // Pool Hourly
+  poolMetricsHourlySnapshot.cumulativeTotalRevenueUSD =
+    pool.cumulativeTotalRevenueUSD;
+  poolMetricsHourlySnapshot.hourlyTotalRevenueUSD = stakingRewards;
+  poolMetricsHourlySnapshot.outputTokenSupply = pool.outputTokenSupply;
+  poolMetricsHourlySnapshot.outputTokenPriceUSD = pool.outputTokenPriceUSD;
+  poolMetricsHourlySnapshot.save();
+
+  // Protocol
+  protocol.cumulativeTotalRevenueUSD = pool.cumulativeTotalRevenueUSD;
+  protocol.save();
+
+  // Financials Daily
+  financialMetrics.cumulativeTotalRevenueUSD = pool.cumulativeTotalRevenueUSD;
+  financialMetrics.dailyTotalRevenueUSD =
+    poolMetricsDailySnapshot.dailyTotalRevenueUSD;
+  financialMetrics.save();
+}
+
+export function updateProtocolSideRevenueMetrics(
+  block: ethereum.Block,
+  amount: BigDecimal
+): void {
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  const protocol = getOrCreateProtocol();
+  const financialMetrics = getOrCreateFinancialDailyMetrics(block);
+  const poolMetricsDailySnapshot = getOrCreatePoolsDailySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+  const poolMetricsHourlySnapshot = getOrCreatePoolsHourlySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+
+  // Staking rewards revenue is in ETH (rebased in stETH for user), price in ETH
+  const amountUSD = amount;
+
+  // Pool
+  pool.cumulativeProtocolSideRevenueUSD = amountUSD;
+  pool.save();
+
+  // Pool Daily
+  poolMetricsDailySnapshot.cumulativeProtocolSideRevenueUSD =
+    pool.cumulativeProtocolSideRevenueUSD;
+  poolMetricsDailySnapshot.dailyProtocolSideRevenueUSD = amountUSD;
+  poolMetricsDailySnapshot.save();
+
+  // Pool Hourly
+  poolMetricsHourlySnapshot.cumulativeProtocolSideRevenueUSD =
+    pool.cumulativeProtocolSideRevenueUSD;
+  poolMetricsHourlySnapshot.hourlyProtocolSideRevenueUSD = amountUSD;
```

Comment:
Same as above: daily and hourly revenues have the same value as cumulativeProtocolRevenue?